src/java.base/share/classes/java/lang/reflect/AccessFlag.java

```diff
@@ -360,14 +360,19 @@ public Set<Location> apply(ClassFileFormatVersion cffv) {
      * major versions 46 through 60, inclusive (JVMS {@jvms 4.6}),
      * corresponding to Java SE 1.2 through 16.
      */
-    STRICT(Modifier.STRICT, true, Location.EMPTY_SET,
+    STRICT(0x0000_0800, true, !ValhallaFeatures.isEnabled() ? Location.EMPTY_SET : Location.SET_FIELD,
              new Function<ClassFileFormatVersion, Set<Location>>() {
                @Override
                public Set<Location> apply(ClassFileFormatVersion cffv) {
-                   return (cffv.compareTo(ClassFileFormatVersion.RELEASE_2)  >= 0 &&
-                           cffv.compareTo(ClassFileFormatVersion.RELEASE_16) <= 0) ?
-                       Location.SET_METHOD:
-                       Location.EMPTY_SET;}
+                   if (ValhallaFeatures.isEnabled() && cffv.compareTo(ClassFileFormatVersion.RELEASE_22) >= 0) {
+                       return Location.SET_FIELD;
+                   } else {
+                       return (cffv.compareTo(ClassFileFormatVersion.RELEASE_2) >= 0 &&
+                               cffv.compareTo(ClassFileFormatVersion.RELEASE_16) <= 0) ?
```

Comment:
We can probably change this ternary operator to be part of the if-else chain to make it clearer.